File: clients/python/model/loot_box.py
from .vec2_double import Vec2Double
from .vec2_double import Vec2Double
from .item import Item
class LootBox:
    def __init__(self, position, size, item):
        self.position = position
        self.size = size
        self.item = item
    @staticmethod
    def read_from(stream):
        position = Vec2Double.read_from(stream)
        size = Vec2Double.read_from(stream)
        item = Item.read_from(stream)
        return LootBox(position, size, item)
    def write_to(self, stream):
        self.position.write_to(stream)
        self.size.write_to(stream)
        self.item.write_to(stream)
    def __repr__(self):
        return "LootBox(" + \
            repr(self.position) + "," + \
            repr(self.size) + "," + \
            repr(self.item) + \
            ")"

File: clients/javascript/model/loot-box.js
const Vec2Double = require('./vec2-double').Vec2Double;
const Item = require('./item').Item;
class LootBox {
    constructor(position, size, item) {
        this.position = position;
        this.size = size;
        this.item = item;
    }
    static async readFrom(stream) {
        let position;
        position = await Vec2Double.readFrom(stream);
        let size;
        size = await Vec2Double.readFrom(stream);
        let item;
        item = await Item.readFrom(stream);
        return new LootBox(position, size, item);
    }
    async writeTo(stream) {
        let position = this.position;
        await position.writeTo(stream);
        let size = this.size;
        await size.writeTo(stream);
        let item = this.item;
        await item.writeTo(stream);
    }
}
module.exports = { LootBox: LootBox }
